lib/codecept.js

```diff
@@ -152,27 +152,72 @@ class Codecept {
    *
    * @param {string} [test]
    */
-  run(test) {
-    const mocha = container.mocha();
-    mocha.files = this.testFiles;
-    if (test) {
-      if (!fsPath.isAbsolute(test)) {
-        test = fsPath.join(global.codecept_dir, test);
+  runOnce(test) {
+    return new Promise((resolve, reject) => {
+      container.createMocha();
+      const mocha = container.mocha();
+      this.testFiles.forEach((file) => {
+        delete require.cache[file];
+      });
+      mocha.files = this.testFiles;
+      if (test) {
+        if (!fsPath.isAbsolute(test)) {
+          test = fsPath.join(global.codecept_dir, test);
+        }
+        mocha.files = mocha.files.filter(t => fsPath.basename(t, '.js') === test || t === test);
+      }
+      try {
+        mocha.run((failures) => {
+          if (failures === 0) {
+            resolve();
+          } else {
+            reject(new Error(`${failures} tests fail`));
+          }
+        });
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+
+  async runTests(test) {
```

Comment:
Please move this code into `command/run-retries.js`, create a custom runner there
If it is possible to not update this file at all - this will be perfect